src/file_upload.rs

```diff
@@ -171,6 +171,19 @@ pub async fn upload_file(
         ContextualError::IoError("Failed to resolve path served by miniserve".to_string(), e)
     })?;
 
+
+    // Disallow paths outside of restricted directories
+    // TODO: Probably not the most rust-ic style...
+    if !conf.restrict_upload_dir.is_empty() {
+        let upl_path = upload_path.clone().into_os_string().into_string().unwrap();
```

Comment:
Fully agree to how it reads.

One option I am seeing is to use PathBuf for the --restrict-upload-dir argument instead of String and then handle it as a path. @svenstaro, would this make sense?